pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -725,9 +704,12 @@ func (p *Pod) validatePodGroupMetadata(r record.EventRecorder, activePods []core
 	if err != nil {
 		return err
 	}
-	if _, err = p.podGroupIndex(groupTotalCount); err != nil {
+
+	_, err = utilpod.ReadUIntFromLabelBelowBound(p.Object(), kueuealpha.PodGroupPodIndexLabel, groupTotalCount-1)
```

Comment:
Why the `-1`?

Reply:
> Why do we want to return error if `uintValue == max`?

In the previous implementation, we were checking for equality.

https://github.com/kubernetes-sigs/kueue/pull/3717/files#diff-dfb49586a8522fa91d733051fd3b7e4b3ff174907898cf249ab16e2620976a5dL572-L574

Reply:
So here it should be -1

Reply:
After this fix: https://github.com/kubernetes-sigs/kueue/pull/3717/files#r1867747724 there should not be a need to pass `-1`

Reply:
It will not work due to https://github.com/kubernetes-sigs/kueue/pull/3717#discussion_r1867688322.

Reply:
Sorry, in my comment I missed the bug in the `ReadUIntFromLabelBelowBound` function, as indicated by @PBundyra

Reply:
I think if we fix `ReadUIntFromLabelBelowBound` then we can just call it without any -1 / +1

Reply:
In that case on topology we need to set +1

Reply:
FIxed